hlapipeline/utils/filter.py

```diff
@@ -0,0 +1,156 @@
+    isSuccess = False
+    #isSuccess = 0
+    dateObs = None
+    namesArray = ('imageName', 'instrument', 'detector', 'filter', 'aperture', 'obstype', 
+            'subarray', 'dateObs', 'doProcess', 'catalog', '# found sources', '# match sources', 
+            'rms_x', 'rms_y', 'isSuccess')
+    dataType = ('S20', 'S20', 'S20', 'S20', 'S20', 'S20', 'b', 'S20', 'b', 'S20', 'i4', 'i4', 'f8', 'f8', 'b')
+
+    outputTable = Table(names=namesArray,dtype=dataType)
+
+    # Loop over the list of images to determine viability for alignment processing
+    for inputFile in inputFileList:
+
+        header_hdu  = 0
+        header_data = fits.getheader(inputFile, header_hdu)
+
+        obstype  = (header_data[OBSKEY]).upper()
+        mtflag   = (header_data[MTKEY]).upper()
+        scan_typ = (header_data[SCNKEY]).upper()
+
+        sfilter  = (header_data[FILKEY]).upper()
+        aperture = (header_data[APKEY]).upper()
+        targname = (header_data[TARKEY]).upper()
+        expotime = header_data[EXPKEY]
+
+        # Keywords to use potentially for analysis
+        instrume = (header_data['INSTRUME']).upper()
+        detector = (header_data['DETECTOR']).upper()
+        subarray = header_data['SUBARRAY']
+        dataObs  = header_data['DATE-OBS']
+
+        # Determine if the image has one of these conditions.  The routine
+        # will exit processing upon the first satisfied condition.
+
+        noProcKey   = None
+        noProcValue = None
+        doProcess = True
+        # Imaging vs spectroscopic or coronagraphic
+        if obstype != 'IMAGING':
+            noProcKey   = OBSKEY
+            noProcValue = obstype 
+
+        # Moving target
+        elif mtflag == 'T':
+            noProcKey   = MTKEY
+            noProcValue = mtflag 
+
+        # Bostrophidon without or with dwell (WFC3 only)
+        elif instrume == 'WFC3' and any ([scan_typ == 'C', scan_typ == 'D']):
+            noProcKey   = SCNKEY
+            noProcValue = scan_typ
+
+        # Filter which begins with !F (e.g., 'POL*' or 'G*')
+        elif sfilter[0] != 'F': 
+            noProcKey   = FILKEY
+            noProcValue = sfilter
+
+        # Ramp, polarizer, or grism 
+        elif any (['RAMP' in aperture, 'POL' in aperture, 'GRISM' in aperture, 
+                  '-REF' in aperture]):
+            noProcKey   = APKEY
+            noProcValue = aperture 
+
+        # Calibration target
+        elif any (['DARK' in targname, 'TUNG' in targname, 'BIAS' in targname, 
+                  'FLAT' in targname, 'DEUT' in targname, 'EARTH-CAL' in targname]):
+            noProcKey   = TARKEY
+            noProcValue = targname
+
+        # Exposure time of zero
+        elif math.isclose(expotime, 0.0, abs_tol=1e-5):
+            noProcKey   = EXPKEY
+            noProcValue = expotime 
+
+        if (noProcKey is not None):
+            doProcess = False
+
+        # Issue message to log file 
+        #issue_msg(inputFileList, noProcKey, noProcValue)
```

Comment:
This output message would be useful to report.  Comments to stdout can always be redirected to a log as needed.